nm: update connection for state:down interface

cathay4t opened the pull request:
When certain interface been marked as `state:down`, nmstate does not
update its connection/profile even desire state requested.

Fixed by storing pending update connections to `nm_conns_to_update`.
Also ignore verification on down physical interface as DHCP and other
kernel state will not match.

Integration test case included.

ffmancera reviewed (changes requested):
This change is breaking ovs support, please could you take a look? Thanks!

cathay4t committed:
nm: update connection for state:down interface

When certain interface been marked as `state:down`, nmstate does not
update its connection/profile even desire state requested.

Fixed by storing pending update connections to `nm_conns_to_update`.
Also ignore verification on down physical interface as DHCP and other
kernel state will not match.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
CI fixed. Please review again. Thanks!
